js/client/modules/@arangodb/arango-query-cursor.js

```diff
@@ -219,7 +221,18 @@ ArangoQueryCursor.prototype.next = function () {
       // load more results
       var requestResult = this._database._connection.POST(this._baseurl(), null);
 
-      arangosh.checkRequestResult(requestResult);
+
+      try {
+        arangosh.checkRequestResult(requestResult);
+      } catch (err) {
+        if (latestBatchId !== undefined) {
+          requestResult = this._database._connection.POST(this._baseurl() + '/' + encodeURIComponent(latestBatchId), null);
+          arangosh.checkRequestResult(requestResult);
+        } else {
+          // throw the error again for the case in which the request to the latest batch is not retryable
+          throw(err);
```

Comment:
`throw` is a keyword, doesn't need parentheses.
```suggestion
          throw err;
```